Model/ModelManager.php

```diff
@@ -385,8 +386,12 @@ public function getNormalizedIdentifier($entity)
             throw new \RunTimeException('Invalid argument, object or null required');
         }
 
-        // the entities is not managed
-        if (!$entity || !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity)) {
+        if (!$entity) {
+            return;
+        }
+
+        $entityState = $this->getEntityManager($entity)->getUnitOfWork()->getEntityState($entity);
+        if (UnitOfWork::STATE_NEW == $entityState || UnitOfWork::STATE_REMOVED == $entityState) {
```

Comment:
All fine by me!

(On reflection, I agree with you about the strict comparison. But not entirely convinced that the `in_array` version is more readable!)

P.S. Dredging my memory for in_array performance, I think it's something to do with the way in_array scans the whole array... but it's almost certainly a non-issue on small arrays like this, it's just that, over the years, I've become hard-coded myself to think twice about using it!